### PR TITLE
fix(annotations): Select newly created annotation

### PR DESCRIPTION
**Pull request by ConradJChan**

### CODE REVIEW

File: src/lib/viewers/image/ImageViewer.js [hunk 1]
```diff
@@ -507,6 +508,22 @@ class ImageViewer extends ImageBaseViewer {
         this.annotator.toggleAnnotationMode(nextMode);
         this.processAnnotationModeChange(nextMode);
     }
+
+    handleAnnotationCreateEvent({ annotation: { id } = {}, meta: { status } = {} }) {
+        // Only on success do we exit create annotation mode. If error occurs,
+        // we remain in create mode
+        if (status === 'success') {
```
Comment (jstoffan) [inline src/lib/viewers/image/ImageViewer.js:515]: Should this be a guard clause?

File: src/lib/viewers/image/ImageViewer.js [hunk 2]
```diff
@@ -507,6 +508,22 @@ class ImageViewer extends ImageBaseViewer {
...
+            this.annotator.emit('annotations_active_set', id);
+        }
+    }
+
+    initAnnotations() {
+        super.initAnnotations();
+
+        if (this.areNewAnnotationsEnabled() && this.annotationControls) {
```
Comment (jstoffan) [inline src/lib/viewers/image/ImageViewer.js:523]: Should we be checking `this.annotator`, instead or as well?

File: src/lib/viewers/image/ImageViewer.js [hunk 3]
```diff
@@ -507,6 +508,22 @@ class ImageViewer extends ImageBaseViewer {
...
+            this.annotator.addListener('annotations_create', this.handleAnnotationCreateEvent);
```
Comment (jstoffan) [inline src/lib/viewers/image/ImageViewer.js:524]: Does this listener need to be removed at some point?

File: src/lib/viewers/image/ImageViewer.js (hunk 2 above)
Comment (ConradJChan) [reply src/lib/viewers/image/ImageViewer.js:523]: `BaseViewer.initAnnotations` already assumes `this.annotator` exists, so it seems safe enough?

File: src/lib/viewers/image/ImageViewer.js (hunk 3 above)
Comment (ConradJChan) [reply src/lib/viewers/image/ImageViewer.js:524]: I don't think so, `BaseViewer.destroy` takes care of it by calling `this.annotator.removeAllListeners()`

File: src/lib/viewers/image/ImageViewer.js (hunk 2 above)
Comment (jstoffan) [reply src/lib/viewers/image/ImageViewer.js:523]: I'm confused why the `this.annotationControls` check is needed, then. It's not referenced in the subsequent code or in the handler.

Comment (ConradJChan) [reply src/lib/viewers/image/ImageViewer.js:525]: Will remove that check since it seems unnecessary

**Review by jstoffan — APPROVED** [review]

**Pull request merged**